homeassistant/components/feedreader.py

```diff
@@ -60,22 +61,46 @@ def _update(self):
             elif len(self._feed.entries) > 0:
                 _LOGGER.debug('Entries available in feed %s', self._url)
                 self._publish_new_entries()
-                self._last_entry_timestamp = \
-                    self._feed.entries[0].published_parsed
             else:
                 self._log_no_entries()
+        _LOGGER.info('Fetch from feed "%s" completed', self._url)
+
+    def _update_and_fire_entry(self, entry):
+        """Update last_entry_timestamp and fire entry."""
+        # We are lucky, `published_parsed` data available,
+        # let's make use of it to publish only new available
+        # entries since the last run
+        if 'published_parsed' in entry:
```

Comment:
Ok, if they have subclassed a dict then it should all work fine. I don't expect tests for components that interact with the outside world but it is just so easy when you're developing something.

Anyway, since it's all supposed to work let's stick with it.